game/src/info/trip.rs

```diff
@@ -386,6 +389,329 @@ pub fn cancelled(ctx: &mut EventCtx, app: &App, id: TripID) -> Widget {
     Widget::col(col)
 }
 
+/// Highlights intersections which were "slow" on the map
+fn highlight_slow_intersections(app: &App, details: &mut Details, id: TripID, ctx: &EventCtx) {
+    let data_opt = &app.primary.sim.get_analytics().trip_intersection_delays(id);
+    if data_opt.is_some() {
+        let data = data_opt.unwrap();
+        for (id, time) in data {
+            let intersection = app.primary.map.get_i(id.parent);
+            // Maybe alter the delay times
+            let (normal_delay_time, slow_delay_time) = if intersection.is_traffic_signal() {
+                (30, 120)
+            } else {
+                (5, 30)
+            };
+            let (fg_color, bg_color) = if time < &normal_delay_time {
+                (Color::WHITE, app.cs.normal_slow_intersection)
+            } else if time < &slow_delay_time {
+                (Color::BLACK, app.cs.slow_intersection)
+            } else {
+                (Color::WHITE, app.cs.very_slow_intersection)
+            };
+            // Draw label after, to prevent being obscured by the line
+            let time_duration = Duration::seconds(*time as f64);
+            let unzoomed_label = Text::from(TextSpan::from(
```

Comment:
Style: It's a matter of style preference, but the intermediate variables feel a bit verbose here. I think adding the `.render(ctx).centered_on(...)` here would be more readable, maybe even directly appending to `detailed.unzoomed` and not giving this a name. But maybe it's easier to detangle what's happening when it's broken down into steps?

Reply:
Nah, it makes sense, it was only because I was going through a stage of changing it frequently, and didn't change it back.